### PR TITLE
AE-2306 level bust

### PR DESCRIPTION
**Pull request by gille02d**

### CODE REVIEW

File: analysis_engine/key_point_values.py [hunk 1]
```diff
@@ -19374,3 +19375,142 @@ def derive(self, alt_descending=KTI('Altitude When Descending'),
...
+        #    altitude. We ignore the missed approach altitude section until the
+        #    end of the approach phase. We then check the previous Altitude
+        #    Selected which is the intercept altitude if we maintained it for at
+        #    least 20 sec. In that case, we mask out the part where we left last
+        #    the intercept altitude.
+        # 3. There was no intercept altitude - the path was intercepted during
+        #    descent. We mask out the missed approach altitude from the start
+        #    until the end of the approach phase. We mask out the previous
+        #    Altitude Selected from the last moment we crossed it.
+        apps = apps or []
```
Comment (kirku1g) [inline analysis_engine/key_point_values.py]: I don't believe the empty list fallback is required as SectionNode inherits from list and is a required dependency, therefore cannot be None.

Comment (gille02d) [reply analysis_engine/key_point_values.py]: Great point. This makes me wonder then if the KPV derive method would be called for a flight without any Approach and Landing - say an incomplete flight. Would an empty `S('Approach And Landing')` be passed in or `None`? In the latter case, it would mean the code would not run, and that's not the desired behaviour.

Either I need to add a `can_operate` method to allow this KPV to run when there are no approaches or I need to remove line 19420 and update the tests making sure I always provide an empty `S('Approach And Landing')`.

Comment (kirku1g) [reply analysis_engine/key_point_values.py]: Since Approach And Landing is a required dependency, a SectionNode (potentially empty) would always be passed. Since dependencies are worked out from parameter availability (and their derivatives), whether or not the Approach And Landing phase is available for this node to process isn't dependent on the Segment Type (perhaps with some minor exclusions, but generally this is the case). Therefore you can expect an empty SectionNode to be passed in rather than writing can_operate methods for these cases.

Comment (gille02d) [reply analysis_engine/key_point_values.py]: Thank you for the explanation. It makes total sense now!

I've updated the code and the tests accordingly.
